File: app/src/main/java/com/future/awaker/imageloader/IImageLoader.java
package com.future.awaker.imageloader;

import android.graphics.drawable.Drawable;
import android.widget.ImageView;

import com.bumptech.glide.request.RequestListener;

/**
 * Copyright ©2017 by ruzhan
 */

public interface IImageLoader {

    void load(ImageView imageView, String url);

    void load(ImageView imageView, String url, RequestListener<Drawable> listener);

    void load(ImageView imageView, int resId, RequestListener<Drawable> listener);

    void loadThumb(ImageView imageView, String url);

    void loadCropCircle(ImageView imageView, String url);

    void loadCropCircle(ImageView imageView, int resId);
}

File: common/src/main/java/com/ruzhan/common/imageloader/IImageLoader.kt
package com.ruzhan.common.imageloader

import android.graphics.drawable.Drawable
import android.widget.ImageView

import com.bumptech.glide.request.RequestListener

interface IImageLoader {

    fun load(imageView: ImageView, url: String)

    fun load(imageView: ImageView, url: String, listener: RequestListener<Drawable>)

    fun load(imageView: ImageView, resId: Int, listener: RequestListener<Drawable>)

    fun loadThumb(imageView: ImageView, url: String)

    fun loadCropCircle(imageView: ImageView, url: String)

    fun loadCropCircle(imageView: ImageView, resId: Int)
}
